plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSharedGlueMetastore.java

```diff
@@ -0,0 +1,161 @@
+import io.trino.plugin.hive.metastore.glue.DefaultGlueColumnStatisticsProviderFactory;
+import io.trino.plugin.hive.metastore.glue.GlueHiveMetastore;
+import io.trino.plugin.hive.metastore.glue.GlueHiveMetastoreConfig;
+import io.trino.plugin.tpch.TpchPlugin;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import io.trino.tpch.TpchTable;
+import org.testng.annotations.AfterClass;
+
+import java.nio.file.Path;
+import java.util.Optional;
+
+import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
+import static io.trino.plugin.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static io.trino.testing.QueryAssertions.copyTpchTables;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static java.lang.String.format;
+
+/**
+ * Tests metadata operations on a schema which has a mix of Hive and Delta Lake tables.
+ * <p>
+ * Requires AWS credentials, which can be provided any way supported by the DefaultProviderChain
+ * See https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default
+ */
+public class TestDeltaLakeSharedGlueMetastore
+        extends BaseDeltaLakeSharedMetastoreTest
+{
+    private static final Logger LOG = Logger.get(TestDeltaLakeSharedGlueMetastore.class);
+
+    private Path dataDirectory;
+    private HiveMetastore glueMetastore;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session deltaLakeSession = testSessionBuilder()
+                .setCatalog("delta")
+                .setSchema(schema)
+                .build();
+        Session hiveSession = testSessionBuilder()
+                .setCatalog("hive")
+                .setSchema(schema)
+                .build();
+
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(deltaLakeSession).build();
+
+        queryRunner.installPlugin(new TpchPlugin());
+        queryRunner.createCatalog("tpch", "tpch");
+
+        this.dataDirectory = queryRunner.getCoordinator().getBaseDataDir().resolve("delta_lake_data");
+        this.dataDirectory.toFile().deleteOnExit();
+
+        queryRunner.installPlugin(new DeltaLakePlugin());
+        queryRunner.createCatalog(
+                "delta",
+                "delta-lake",
+                ImmutableMap.<String, String>builder()
+                        .put("hive.metastore", "glue")
+                        .put("hive.metastore.glue.default-warehouse-dir", dataDirectory.toString())
+                        .buildOrThrow());
+        queryRunner.createCatalog(
+                "delta_with_redirections",
+                "delta-lake",
+                ImmutableMap.<String, String>builder()
+                        .put("hive.metastore", "glue")
+                        .put("hive.metastore.glue.default-warehouse-dir", dataDirectory.toString())
+                        .put("delta.hive-catalog-name", "hive")
```

Comment:
In case it is considered needed I could add an extra Glue test class containing `delta.hide-non-delta-lake-tables` property for the Delta Lake connectors.